packages/block-editor/src/components/use-consolidated-styles/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * // TODO This is an example targeted at specific style properties. We could consolidate at the highest level, e.g., return a merged styles object for all in packages/block-editor/src/hooks/style.js
+ * Hook that retrieves consolidated styles from the block editor store,
+ * and merges them with incoming user styles for a particular property.
+ *
+ * @param  {Object} userStyles    User-selected styles from the editor.
+ * @param  {string} styleProperty Targets a specific property. If not passed then we return the entire merged object.
+ *
+ * @return {Object}               The merged user styles, or original user styles if no consolidated styles.
+ *
+ * @example
+ * ```js
+ * const consolidatedStyles = useConsolidatedStyles( props.attributes?.style, 'border' );
+ * ```
+ */
+export default function useConsolidatedStyles( userStyles, styleProperty ) {
+	const { name: blockName } = useBlockEditContext();
+
+	const consolidatedStyles = useSelect(
+		( select ) => {
+			const { getSettings } = select( blockEditorStore );
+			const consolidatedBlockStyles =
+				getSettings().__experimentalStyles?.blocks || {};
+
+			return consolidatedBlockStyles &&
+				consolidatedBlockStyles[ blockName ]
+				? consolidatedBlockStyles[ blockName ]
+				: undefined;
+		},
+		[ blockName, styleProperty ]
+	);
+
+	if ( ! userStyles || ! styleProperty ) {
+		return userStyles;
+	}
+
+	if ( consolidatedStyles ) {
+		return {
+			...userStyles,
+			[ styleProperty ]: {
+				...consolidatedStyles[ styleProperty ],
+				...userStyles[ styleProperty ],
+			},
```

Comment:
This doesn't take into account that some values will be `var('--some-var-name')`. 

Nor have I asked the question whether we should at all.

If yes, we could try pre-filling with `getPropertyValue('--some-var-name')`.

Reply:
> If yes, we could see if pre-filling with getPropertyValue('--some-var-name') is possible.

Turns out we can access CSS var values in presets: https://github.com/WordPress/gutenberg/pull/34178/files#diff-c7f239b002e1dfa2d70dc9ac84af4975a6c1cd63d00f0c14e4510c5a6a9ecc93